test-utils/src/dev/clickhouse.rs

```diff
@@ -108,7 +208,7 @@ impl ClickHouseDeployment {
     pub async fn cleanup(&mut self) -> Result<(), anyhow::Error> {
         match self {
             ClickHouseDeployment::SingleNode(instance) => {
-                instance.cleanup().await
+                instance.process.cleanup().await
```

Comment:
Sort of. The `cleanup()` method takes `&mut self`, and I was trying to avoid implementing `DerefMut` so that you can't modify the processes except through the `ClickHouseDeployment` type. It's not crucial, since this is a test type, but maybe we can wait until we really need to test things like killing individual nodes in the cluster?

Reply:
Ah! Yeah, that makes sense. +1 on not implementing `DerefMut` if we don't really need it for now.

Reply:
Sounds good. I've added a comment explaining that choice at this line to help.

Reply:
Commented in efb0adbb0